ebmlite/encoding.py

```diff
@@ -124,6 +124,7 @@ def encodeUInt(val, length=None):
             left-padded with ``0x00`` if `length` is not `None`.
         @raise ValueError: raised if val is longer than length.
     """
+    val = int(val)
     pad = b'\x00'
     packed = _struct_uint64.pack(val).lstrip(pad) or pad
```

Comment:
I think that raising an exception would be "expected behavior" unless the float was (approximately) equal to an integer.  We should probably generate some kind of warning if we're going to change that.  For example:

```suggestion
    if not isinstance(val, int):
        if val % 1 != 0.:
            warnings.warn('{} is not close to an integer, {} will be rounded off'.format(val, val - int(val)))
        val = int(val)
    pad = b'\x00'
    packed = _struct_uint64.pack(val).lstrip(pad) or pad
```

Reply:
Should we assume the remainder is zero, or just below a threshold in order to avoid rounding errors?

Reply:
Python is very good about accounting for rounding error, but since it's just a warning and not an error I'd just check equality rather than thresholding.